src/main/java/uk/gov/hmcts/reform/userprofileapi/controller/UserProfileController.java

```diff
@@ -265,12 +259,22 @@
             consumes = APPLICATION_JSON_UTF8_VALUE,
             produces = APPLICATION_JSON_UTF8_VALUE
     )
+
     @ResponseBody
-    public ResponseEntity<CreateUserProfileResponse> updateUserProfile(@Valid @RequestBody UpdateUserProfileData updateUserProfileData, @PathVariable String userId) {
+    public ResponseEntity<UserProfileRolesResponse> updateUserProfile(@Valid @RequestBody UpdateUserProfileData updateUserProfileData, @PathVariable String userId
+    ) {
```

Comment:
Super hard to read. This is very hard-to-read and confusing code. Consider refactoring.

Reply:
Removed @Valid annotation from method declaration